### ISSUE TITLE
RunningStats.update overwrites the variance accumulator (observation std decays as sigma/sqrt(k))

### ISSUE THREAD

**Issue by JingzheShi** [akanazawa/fpo#13]
### Summary

`RunningStats.update` in `playground/src/flow_policy/math_utils.py` **overwrites** the
second-central-moment accumulator instead of accumulating it, so the observation
normalizer's reported `std` shrinks as `sigma / sqrt(k)` after `k` batches rather than
converging to `sigma`.

The class docstring says "Running statistics using Welford's method", and the expression
itself is the correct batch form — only the `self.var_sum +` prefix is missing.

### Location

https://github.com/akanazawa/fpo/blob/main/playground/src/flow_policy/math_utils.py#L70-L72

```python
new_var_sum = jnp.sum(
    diff_to_old_mean * (x - new_mean), axis=range(batch_ndims)
)
```

### Why this looks unintended rather than deliberate

- The batch identity `M2_new = M2_old + sum_i (x_i - mu_old)(x_i - mu_new)` is exact, so
  the summand is right and only the running term is dropped.
- `self.var_sum` is stored on every update but is **never read for its value** anywhere in
  the repository — the only read is `.shape` in the assert on line 63. A state field that
  is written each step and never consumed is what you would expect if an accumulation was
  intended.

### Minimal reproduction (pure NumPy, no JAX required)

```python
import numpy as np

def step(count, mean, var_sum, x, accumulate):
    new_count = count + x.shape[0]
    diff_to_old_mean = x - mean
    new_mean = mean + diff_to_old_mean.sum(0) / new_count
    term = (diff_to_old_mean * (x - new_mean)).sum(0)
    new_var_sum = (var_sum + term) if accumulate else term   # <-- only difference
    var_clipped = np.clip(new_var_sum / new_count, 1e-12, 1e12)
    return new_count, new_mean, new_var_sum, np.sqrt(var_clipped)

rng = np.random.default_rng(0)
B, D, TRUE_SIGMA = 2048, 4, 3.0
state = {a: (0.0, np.zeros(D), np.zeros(D)) for a in (False, True)}
for k in range(1, 61):
    x = rng.normal(0.0, TRUE_SIGMA, (B, D))
    out = {}
    for acc in (False, True):
        c, m, v = state[acc]
        c, m, v, std = step(c, m, v, x, acc)
        state[acc] = (c, m, v); out[acc] = std.mean()
    if k in (1, 2, 4, 8, 16, 32, 60):
        print(f"{k:4d}  as-written {out[False]:.4f}   accumulated {out[True]:.4f}   sigma/sqrt(k) {TRUE_SIGMA/np.sqrt(k):.4f}")
```

Output (stationary `N(0, 3^2)` input):

```
   1  as-written 3.0019   accumulated 3.0019   sigma/sqrt(k) 3.0000
   2  as-written 2.1029   accumulated 2.9882   sigma/sqrt(k) 2.1213
   4  as-written 1.5021   accumulated 2.9932   sigma/sqrt(k) 1.5000
   8  as-written 1.0517   accumulated 2.9982   sigma/sqrt(k) 1.0607
  16  as-written 0.7528   accumulated 3.0039   sigma/sqrt(k) 0.7500
  32  as-written 0.5347   accumulated 3.0034   sigma/sqrt(k) 0.5303
  60  as-written 0.3900   accumulated 3.0032   sigma/sqrt(k) 0.3873
```

The as-written column tracks `sigma / sqrt(k)`; normalized observations are correspondingly
scaled up by roughly `sqrt(k)`.

### Possible fix

Minimal, matching the existing expression:

```python
new_var_sum = self.var_sum + jnp.sum(
    diff_to_old_mean * (x - new_mean), axis=range(batch_ndims)
)
```

Or Chan's parallel form, which avoids relying on the cross term when batch means drift:

```python
nb = onp.prod(x.shape[:batch_ndims])
batch_mean = jnp.mean(x, axis=range(batch_ndims))
batch_var_sum = jnp.sum((x - batch_mean) ** 2, axis=range(batch_ndims))
delta = batch_mean - self.mean
new_count = self.count + nb
new_var_sum = self.var_sum + batch_var_sum + delta ** 2 * (self.count * nb / new_count)
new_mean = self.mean + delta * nb / new_count
```

Both give identical results on the reproduction above.

### Separate, minor

`count` is `jnp.float32` (line 52). float32 represents integers exactly only up to
`2**24 = 16,777,216`; a 60M-step run exceeds that, so `count` itself becomes inexact.
Independent of the above and much smaller in effect, but it touches the same update.

### Context

Observed while running the playground FPO code in our own fork. In our environment (a
DM-Control task, 60M steps, ~61 outer blocks) we see occasional mid-training value-function
blow-ups; replaying a failing block with identical parameters, data, optimizer state and RNG
and changing only this one term moved that block's mean value loss from `2.2e10` to `24.78`.
**That is an observation from our setup, not a claim about results produced with this
repository** — we mention it only as motivation for reporting.

Issue #4 ("Reward=nan when training HumanoidWalk in playground") describes a superficially
similar pattern of training that is fine early and degenerates later. We have not verified
any connection and are not asserting one; noting it only in case it is useful.

Happy to provide further detail. Not opening a PR since the appropriate variant is your call.
